sdk/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -511,124 +506,75 @@ public void setAutoCaptureSessions(boolean autoCapture) {
     }
 
     /**
-     * Set details of the user currently using your application.
-     * You can search for this information in your Bugsnag dashboard.
-     * <p/>
-     * For example:
-     * <p/>
-     * client.setUser("12345", "james@example.com", "James Smith");
-     *
-     * @param id    a unique identifier of the current user (defaults to a unique id)
-     * @param email the email address of the current user
-     * @param name  the name of the current user
-     */
-    public void setUser(String id, String email, String name) {
-        setUserId(id);
-        setUserEmail(email);
-        setUserName(name);
-    }
-
-    /**
-     * Removes the current user data and sets it back to defaults
+     * @return the current user information associated with the Bugsnag {@link Client}
      */
-    public void clearUser() {
-        user.setId(deviceData.getUserId());
-        user.setEmail(null);
-        user.setName(null);
-
-        SharedPreferences sharedPref =
-            appContext.getSharedPreferences(SHARED_PREF_KEY, Context.MODE_PRIVATE);
-        sharedPref.edit()
-            .remove(USER_ID_KEY)
-            .remove(USER_EMAIL_KEY)
-            .remove(USER_NAME_KEY)
-            .apply();
-        notifyBugsnagObservers(NotifyType.USER);
+    public User getUser() {
+        return user;
     }
 
     /**
-     * Set a unique identifier for the user currently using your application.
-     * By default, this will be an automatically generated unique id
-     * You can search for this information in your Bugsnag dashboard.
+     * Set the user information that the Bugsnag {@link Client} supplies with every error.<p>
+     * Pass NULL to clear the current user data, i.e. reset it to defaults.
      *
-     * @param id a unique identifier of the current user
+     * @param user a {@link User} object made by {@link User#builder()} or NULL
      */
-    public void setUserId(String id) {
-        setUserId(id, true);
+    public void setUser(@Nullable User user) {
```

Comment:
The JavaDocs for this method should include as much information as before, although we'll need to change how the params are handled:

```
Set details of the user currently using your application.
-     * You can search for this information in your Bugsnag dashboard.
-     * <p/>
-     * For example:
-     * <p/>
-     * client.setUser("12345", "james@example.com", "James Smith");
-     *
-     * @param id    a unique identifier of the current user (defaults to a unique id)
-     * @param email the email address of the current user
-     * @param name  the name of the current user
```

Reply:
That information is available in the javadoc of `User` though? Which the dev will see when they construct a `User` object. By duplicating java doc here, everytime `User` is changed, the java doc has to be updated in 3 places instead of just in `User` where it belongs.